Mathlib/RingTheory/LocalRing/Basic.lean

```diff
@@ -31,6 +31,13 @@ theorem of_nonunits_add [Nontrivial R]
     (h : ∀ a b : R, a ∈ nonunits R → b ∈ nonunits R → a + b ∈ nonunits R) : LocalRing R :=
   ⟨fun {a b} hab => or_iff_not_and_not.2 fun H => h a b H.1 H.2 <| hab.symm ▸ isUnit_one⟩
 
+
+end Semiring
+
+section CommSemiring
+
+variable [CommSemiring R]
+
 /-- A semiring is local if it has a unique maximal ideal. -/
 theorem of_unique_max_ideal (h : ∃! I : Ideal R, I.IsMaximal) : LocalRing R :=
```

Comment:
It's interesting that this holds also for `Ring`s, which inspired me to ask a question: https://math.stackexchange.com/questions/5004383/semiring-with-a-unique-left-ideal